### PR TITLE
Feature/cljs commands

### PR DESCRIPTION
**Pull request by janherich**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Finished move to cljs commands.

Some highlights:

* Overall, huge amounts of code removed, whole namespaces like `status-im.commands`, `status-im.bots` a lot of event namespaces, etc.
* Massive reduction in complexity of core chat namespaces like `status-im.chat.events.input` (~500LOC -> ~130LOC) and `status-im.chat.models.message` (~550LOC -> ~300LOC)
* Generic message handling code is much cleaner and nicer, easier to understand because it's not burdened with implementation of particular commands, that's true for views, events, models
* Everything related to commands is nicely encapsulated behind protocol API (`status-im.chat.commands.protocol`)
* Adding new commands with even complicated functionality should be a breeze, with minimal chance of breaking the existing ones
* More comprehensive unit tests exercising functionality which could not be captured in tests at all before the changes (behaviour of particular commands)
* Console removed altogether (so requesting test ETH/Tokens works the same way through status DApp)

### Testing notes (optional):
Comprehensive testing of chat functionality & commands

status: ready

### CODE REVIEW

**Review by rasom — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]
Wow! impressive work!

**Comment by flexsurfer** [comment]
@janherich but you didn't remove js files and JSC in this PR?

**Comment by jeluard** [comment]
Huge! What should we focus on for the review?

**Review by dmitryn — APPROVED** [review]

**Comment by janherich** [comment]
@flexsurfer No, I just removed usage of it, but JSC is still included and bundled into the app, there are also ~1000LOC of js code (bots + api) ripe for throwing away.
But the PR is big as it's now, I will remove all those additional things in next PR (which will be already super simple as it will be just deleting unused code and nothing more).

@jeluard I would suggest focusing on the `status-im.chat.commands` namespace (and all sub-namespaces), the protocol, etc.

File: src/status_im/chat/commands/impl/transactions.cljs [hunk 1]
```diff
@@ -173,36 +177,32 @@
...
 
+(defn- inject-network-price-info [{:keys [amount asset] :as parameters} {:keys [db]}]
+  (let [{:keys [chain prices]} db
+        currency               (-> db
+                                   (get-in [:account/account :settings :wallet :currency] :usd)
+                                   name
+                                   string/upper-case)]
+    (assoc parameters
+           :network     chain
+           ;; TODO(janherich) - shouldn't this be rather computed on the receiver side ?
```
Comment (goranjovic) [inline src/status_im/chat/commands/impl/transactions.cljs:190]: Yes, it should. We have an issue for that after this one is merged.

File: src/status_im/chat/commands/protocol.cljs [hunk 2]
```diff
@@ -29,21 +29,36 @@
...
     "Function rendering the short-preview of the command message, used when
     displaying the last message in list of chats on home tab.
     There is no argument names `parameters` anymore, as the message object
     contains everything needed for short-preview/preview to render.")
-  (preview [this command-message cofx]
+  (preview [this command-message]
     "Function rendering preview of the command message in message stream"))
+
+(defprotocol Yielding
+  "Protocol for defining commands yielding control back to application during the send flow"
```
Comment (goranjovic) [inline src/status_im/chat/commands/protocol.cljs:47]: Maybe rename `send` -> `command` in the comment here.  I know it's only actually used for `/send` now, but just to keep it generic.

**Review by goranjovic — APPROVED** [review]

**Comment by goranjovic** [comment]
Just love these PRs with lots of red. The next one is going to be awesome then

**Comment by jakubgs** [comment]
Please merge this fix to get Jenkins PR builds fixed: https://github.com/status-im/status-react/commit/0e502ac5b810e5fe2066db8241789e43f4e3a3af

File: src/status_im/chat/commands/protocol.cljs (hunk 2 above)
Comment (janherich) [reply src/status_im/chat/commands/protocol.cljs:47]: It's meant as "command (any) send flow", with send flow being validation/yielding-control/sending-message. 
But it's little bit confusing, I agree :)

Comment (goranjovic) [reply src/status_im/chat/commands/protocol.cljs:47]: Yeah, I get it. "send" as in "send command message". Yeah, confusing.

**Comment by lukaszfryc** [comment]
Hey @janherich, I've found following issues:

1. There is no Console chat when I enable dev mode
2. "Sent" button in transaction request is disabled
- 1st device: "/request" ETH
- 2nd device: wait for the request message. When it appears, the "Sent" button is disabled (grayed out)  
3. Color and text of received "/request" command

When "/request" is received, the last message in Home says "Send transaction 1 ETH" and has black color:

In last nightly build, the text says for instance "Requesting 3 ETH" and is gray:


4. Is this change backward compatible? When I open a message that was sent from this PR to nightly I get an error:

**Comment by janherich** [comment]
Hi @lukaszfryc 
1. No Console is expected, console along with jail is deprecated and as @flexsurfer already implemented the alternative to `/faucet` functionality in the status DApp
I will fix the rest of the issues immediately, and yes, the PR build should be 100% compatible with previous builds.

**Comment by yevh-berdnyk** [comment]
Hey @janherich , is there a way to leave commands in 1-1 chat with their accessibility ids? (`request-payment-button` and `send-payment-button`)

And here are the issues from autotests:
3 tests failed sending funds (STT and ETH) from wallet with `chat.chat-id must of type string, got null`:
image

Error `Max number of decimals is 18` is shown in 1-1 chat if requesting ETH from wallet, but number of decimals is less than 18:
image

**Comment by janherich** [comment]
@yevh-berdnyk sure, I will leave the accessibility labels there, I just changed them to `request-button` and `send-button` so they can be automatically generated from command name (and not every command is payment), but I can revert that back.

**Comment by yevh-berdnyk** [comment]
@janherich don't revert then, I'll update ids in auto tests, thanks

**Comment by janherich** [comment]
@yevh-berdnyk Awesome, thanks !

**Comment by janherich** [comment]
@lukaszfryc New build with fixed issues 3. and 4. is on the way, I'm still replicating issue 2.

**Comment by janherich** [comment]
@lukaszfryc All above issues are already fixed, ready to be tested as soon as new build appears :)

**Comment by lukaszfryc** [comment]
@janherich I checked my issues and the one @yevh-berdnyk reported and here are results:

1. When I "/request" or "/send" tokens from this PR to 0.9.22, the messages do not appear on 0.9.22 chat. The transactions are in Wallet Transaction History but do not appear in the chat. If sent from 0.9.22 to this PR, everything works fine. 

2. “Send” button should become disabled after user click it and send the transaction

3. There is no error in chat when number of decimals is exceeded for a token

**Comment by janherich** [comment]
@lukaszfryc again, all the errors above should fixed. I tested extensively 2. and 3. and they are working, the first case I tested only with `0.9.21` <-> PR version, but it should be working with all recent versions (after beta at least).

**Comment by lukaszfryc** [comment]
@janherich works good, please merge 👍 

I tested:
- Upgrade from 0.9.22 and 0.9.23
- Re-tested issues
- Tested messages, notifications
- Checked transactions in wallet and dapps

**Pull request merged**